src/runtime/optional.h

```diff
@@ -64,41 +64,54 @@ class optional {
     void emplace(Args&&... args) {
         if (m_some)
             m_value.~T();
-        m_some = true;
         new (&m_value) T(args...);
+        m_some = true;
```

Comment:
In theory the old version was not exception safe, as if `new T` threw `~optional()` would then incorrectly try to call the destructor on an object that was never constructed.